wal_e/blobstore/swift/utils.py

```diff
@@ -34,7 +34,8 @@ def uri_put_file(creds, uri, fp, content_type=None):
     conn = calling_format.connect(creds)
 
     conn.put_object(
-        container_name, url_tup.path, fp, content_type=content_type
```

Comment:
This looks good to me. I tested some multiple ways and couldn't find an error. If you wish I could test some other scenarios, too.

| File | Content | "Version" |
| --- | -------- | ---------- |
| WALE_SWIFT_PREFIX | `swift://thomas_test` | 1 |
| WALE_SWIFT_PREFIX | `swift://thomas_test/` | 1 |

# Backup-Push
## Result - Version 1
```
wal_e.worker.upload INFO     MSG: begin uploading a base backup volume
        DETAIL: Uploading to "swift://thomas_test/basebackups_005/base_000000010000000000000004_00000040/tar_partitions/part_00000000.tar.lzo".
        STRUCTURED: time=2017-06-20T09:25:07.403117-00 pid=28820
DEBUG: url_tup.path == /basebackups_005/base_000000010000000000000004_00000040/tar_partitions/part_00000000.tar.lzo
DEBUG: url_tup.path.lstrip('/') == basebackups_005/base_000000010000000000000004_00000040/tar_partitions/part_00000000.tar.lzo
wal_e.worker.upload INFO     MSG: finish uploading a base backup volume
```

## Result - Version 2

```
wal_e.worker.upload INFO     MSG: begin uploading a base backup volume
        DETAIL: Uploading to "swift://thomas_test/basebackups_005/base_000000010000000000000008_00000040/tar_partitions/part_00000000.tar.lzo".
        STRUCTURED: time=2017-06-20T09:27:39.765181-00 pid=28900
DEBUG: url_tup.path == /basebackups_005/base_000000010000000000000008_00000040/tar_partitions/part_00000000.tar.lzo
DEBUG: url_tup.path.lstrip('/') == basebackups_005/base_000000010000000000000008_00000040/tar_partitions/part_00000000.tar.lzo
wal_e.worker.upload INFO     MSG: finish uploading a base backup volume
```

# Backup-Fetch

## Result - Version 1

```
wal_e.main   INFO     MSG: starting WAL-E
        DETAIL: The subcommand is "backup-fetch".
        STRUCTURED: time=2017-06-20T09:29:48.667641-00 pid=28947
wal_e.worker.swift.swift_worker INFO     MSG: beginning partition download
        DETAIL: The partition being downloaded is part_00000000.tar.lzo.
        HINT: The absolute Swift object name is basebackups_005/base_000000010000000000000008_00000040/tar_partitions/part_00000000.tar.lzo.
        STRUCTURED: time=2017-06-20T09:29:49.351235-00 pid=28947
```

## Result - Version 2

```
wal_e.main   INFO     MSG: starting WAL-E
        DETAIL: The subcommand is "backup-fetch".
        STRUCTURED: time=2017-06-20T09:30:36.907285-00 pid=28958
wal_e.worker.swift.swift_worker INFO     MSG: beginning partition download
        DETAIL: The partition being downloaded is part_00000000.tar.lzo.
        HINT: The absolute Swift object name is basebackups_005/base_000000010000000000000008_00000040/tar_partitions/part_00000000.tar.lzo.
        STRUCTURED: time=2017-06-20T09:30:37.446041-00 pid=28958
```